sriov/sriov.go

```diff
@@ -37,17 +38,18 @@ func setupVF(conf *SriovConf, ifName string, netns ns.NetNS) error {
 
 	if args.VF != 0 {
 		vfIdx = int(args.VF)
-		vfDevName, err = getVFDeviceName(masterName, vfIdx)
+		vfDevName, locker, err = getVFDeviceName(masterName, vfIdx)
 		if err != nil {
 			return err
 		}
 	} else {
 		// alloc a free virtual function
-		if vfIdx, vfDevName, err = allocFreeVF(masterName); err != nil {
+		if vfIdx, vfDevName, locker, err = allocFreeVF(masterName); err != nil {
```

Comment:
Maybe have no need to change `allocFreeVF` function, We can change it as follows:

```
locker.Lock()
allocFreeVF()
locker.UnLock()
```

Reply:
yes, It's more better implementation. locker is global, Lock() maybe need an angument vfId and make it saved, Unlock() free the vfId and opened file. I will try